shipment_advice_planner/wizards/shipment_advice_planner.py

```diff
@@ -0,0 +1,152 @@
+    def _onchange_check_warehouse(self):
+        self.ensure_one()
+        self._check_warehouse()
+
+    @api.constrains("picking_to_plan_ids")
+    def _check_picking_to_plan(self):
+        for rec in self:
+            if rec.picking_to_plan_ids.filtered(
+                lambda p: not p.can_be_planned_in_shipment_advice
+            ):
+                raise ValidationError(
+                    _("The transfers selected must be ready and of the delivery type.")
+                )
+
+    @api.onchange("picking_to_plan_ids")
+    def _onchange_check_picking_to_plan(self):
+        self.ensure_one()
+        self._check_picking_to_plan()
+
+    @api.model
+    def _get_compute_picking_to_plan_ids_depends(self):
+        return ["shipment_planning_method", "warehouse_id"]
+
+    @api.depends(lambda m: m._get_compute_picking_to_plan_ids_depends())
+    def _compute_picking_to_plan_ids(self):
+        """meant to be inherited if for specific method picking to plan should have
+        a default record set"""
+        active_model = self.env.context.get("active_model")
+        active_ids = self.env.context.get("active_ids")
+        if not active_ids or active_model != "stock.picking":
+            return self.update({"picking_to_plan_ids": False})
+        pickings_to_plan = self.env[active_model].search(
+            [
+                ("id", "in", active_ids),
+                ("can_be_planned_in_shipment_advice", "=", True),
+                ("picking_type_id.warehouse_id", "=?", self.warehouse_id.id),
+            ]
+        )
+
+        return self.update({"picking_to_plan_ids": [Command.set(pickings_to_plan.ids)]})
+
+    def button_plan_shipments(self):
+        self.ensure_one()
+        shipment_advices = self._plan_shipments_for_method()
+        if not shipment_advices:
+            return {}
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Shipment Advice"),
+            "view_mode": "tree,form",
+            "res_model": shipment_advices._name,
+            "domain": [("id", "in", shipment_advices.ids)],
+            "context": self.env.context,
+        }
+
+    def _plan_shipments_for_method(self):
+        self.ensure_one()
+        prepare_method_name = self._get_prepare_method_name()
+        if not hasattr(self, prepare_method_name):
+            raise NotImplementedError(
+                _("There is no implementation for the planning method '%s'")
+                % self.shipment_planning_method
+            )
+        prepare_method = getattr(self, prepare_method_name)
+        shipment_advice_model = self.env["shipment.advice"]
+        create_vals = []
+        for (
+            picking_type,
+            pickings_to_plan,
+        ) in self._get_picking_to_plan_by_picking_type().items():
+            create_vals.extend(prepare_method(picking_type, pickings_to_plan))
+        return shipment_advice_model.create(create_vals)
```

Comment:
The generated shipment advice is not visible by default as there is no "arrival date" and it filters on 'Today" by default.
I would add a default now on the arrival date on the `shipment_advice` module.

Reply:
and also on the departure_date (same as arrival_date) otherwise you don't see it in the calendar